src/cmd/top/joblist_pane.c

```diff
@@ -60,21 +93,32 @@ void joblist_pane_draw (struct joblist_pane *joblist)
         double t_run;
 
         if (json_unpack (job,
-                         "{s:I s:i s:i s:s s:i s:i s:f}",
+                         "{s:I s:i s:i s:s s:i s:i s:f s?{s?{s?s}}}",
                          "id", &id,
                          "userid", &userid,
                          "state", &state,
                          "name", &name,
                          "nnodes", &nnodes,
                          "ntasks", &ntasks,
-                         "t_run", &t_run) < 0)
+                         "t_run", &t_run,
+                         "annotations",
+                           "user",
+                             "uri", &uri) < 0)
             fatal (errno, "error decoding a job record from job-list RPC");
```

Comment:
`json_unpack()` won't reliably set errno on failure

Reply:
I don't think that line was part of this PR, but I'll add a commit to fix it.